pkg/cloud/services/network/subnets.go

```diff
@@ -468,7 +472,10 @@ func (s *Service) createSubnet(sn *infrav1.SubnetSpec) (*infrav1.SubnetSpec, err
 	}
 
 	subnet := &infrav1.SubnetSpec{
-		ID:               *out.Subnet.SubnetId,
+		// Preserve the original identifier, this won't be a valid subnet ID, which
+		// is stored in the ResourceID field.
+		ID:               sn.ID,
+		ResourceID:       *out.Subnet.SubnetId,
```

Comment:
Q: Here we are setting ResourceID to SubnetId and in l.340 to SubnetArn, is this the same thing?

(just to be sure)

Reply:
Yes

Reply:
I guess this got fixed and we're not using ARNs in `ResourceID` anymore? See my other review comment where that is confusing in a leftover code comment.